packages/lib/server/getLuckyUser.ts

```diff
@@ -570,6 +671,63 @@ async function fetchAllDataNeededForCalculations<
     }),
   ]);
 
+  const userFullDayBusyTimes = new Map<number, { start: Date; end: Date }[]>();
+
+  currentMonthUserBusyTimes.forEach((userBusyTime) => {
+    const fullDayBusyTimes = userBusyTime.busyTimes
+      .filter((busyTime) => {
+        const startDate = new Date(busyTime.start);
+        const endDate = new Date(busyTime.end);
+        return endDate.getMonth() === new Date().getMonth() && isFullDayEvent(startDate, endDate);
+      })
+      .map((busyTime) => ({
+        start: new Date(busyTime.start),
+        end: new Date(busyTime.end),
+      }));
+
+    userFullDayBusyTimes.set(userBusyTime.userId, fullDayBusyTimes);
+  });
+
+  // get cal.com OOO data
+  const oooEntries = await prisma.outOfOfficeEntry.findMany({
+    where: {
+      userId: {
+        in: allRRHosts.map((host) => host.user.id),
+      },
+      end: {
+        lte: new Date(),
+        gte: startOfMonth,
+      },
+    },
+    select: {
+      start: true,
+      end: true,
+      userId: true,
+    },
+  });
+
+  const groupedOOOEntries = new Map<number, { start: Date; end: Date }[]>();
```

Comment:
maybe `oooEntriesGroupedByUserId` to be more explicit about what it's grouping - saves scanning the further code.